executor/aggfuncs/func_avg.go

```diff
@@ -72,7 +73,10 @@ func (e *baseAvgDecimal) AppendFinalResult2Chunk(sctx sessionctx.Context, pr Par
 	if err != nil {
 		return err
 	}
-	err = finalResult.Round(finalResult, e.frac, types.ModeHalfEven)
+	if e.retTp == nil {
+		return errors.New("e.retTp of avg should not be nil")
+	}
+	err = finalResult.Round(finalResult, e.retTp.Decimal, types.ModeHalfEven)
```

Comment:
If `e.retTp.Decimal == -1`, before this change, the final result will be rounded with scale = MaxDecimalScale, after this change, the final result will be rounded with scale = -1, it this the expected behavior?

Reply:
fixed